### ISSUE TITLE
trait_duplication_in_bounds: does not differenciate between trait and const trait  (`const_trait_impl`)

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9915]
### Summary

.

### Lint Name

trait_duplication_in_bounds

### Reproducer

I tried this code:

```rust
#![feature(const_trait_impl)]
// This duplicate bound should not result in ambiguities. It should be equivalent to a single ~const
// bound.
const fn equals_self<T: PartialEq + ~const PartialEq>(t: &T) -> bool {
    *t == *t
}

trait A: PartialEq {}
impl<T: PartialEq> A for T {}

fn main() {}
```

I saw this happen:
`cargo clippy --fix -- -Aclippy::all -Wclippy::trait_duplication_in_bounds`
```
    Checking clpy v0.1.0 (/tmp/clpy)
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0277]: can't compare `T` with `_` in const contexts
 --> src/main.rs:5:8
  |
5 |     *t == *t
  |        ^^ no implementation for `T == _`
  |
note: the trait `std::cmp::PartialEq<_>` is implemented for `T`, but that implementation is not `const`
 --> src/main.rs:5:8
  |
5 |     *t == *t
  |        ^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
Original diagnostics will follow.

warning: function `equals_self` is never used
 --> src/main.rs:4:10
  |
4 | const fn equals_self<T: PartialEq + ~const PartialEq>(t: &T) -> bool {
  |          ^^^^^^^^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: these bounds contain repeated elements
 --> src/main.rs:4:25
  |
4 | const fn equals_self<T: PartialEq + ~const PartialEq>(t: &T) -> bool {
  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `PartialEq`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#trait_duplication_in_bounds
  = note: requested on the command line with `-W clippy::trait-duplication-in-bounds`

warning: `clpy` (bin "clpy") generated 2 warnings (run `cargo fix --bin "clpy"` to apply 1 suggestion)
warning: `clpy` (bin "clpy" test) generated 2 warnings (2 duplicates)
    Finished dev [unoptimized + debuginfo] target(s) in 0.35s
```

I expected to see this happen:

### Version

```text
binary: rustc
commit-hash: c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b
commit-date: 2022-11-19
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
